File: vite-vue-ts-frontend/src/modules/task-statuses/types/dto.ts
import type {
  PagerRequest,
  PagerResponse,
} from "../../../shared/types/dto/pager";
import type { Order } from "../../../shared/types/dto/order";
import type { StatusFlags } from "../../../shared/types/status-flags";

export type AddRequest = {
  name: string;
  hexColor: string;
  index: number;
  flags: StatusFlags;
};

export type UpdateRequest = {
  id: string;
  name: string;
  hexColor: string;
  index: number;
  flags: StatusFlags;
};

type SearchRequestFilter = {
  name?: string;
};

export type SearchRequest = {
  pager: PagerRequest;
  order: Order;
  filter?: SearchRequestFilter;
};

export type TaskStatusResponse = {
  id: string;
  name: string;
  hexColor: string;
  index: number;
  flags: StatusFlags;
};

export type SearchResponse = {
  taskStatuses: TaskStatusResponse[];
  pager: PagerResponse;
};

File: golang-backend/internal/handlers/taskstatushandler/dto.go
package taskstatushandler

import "github.com/aportela/doneo/internal/handlers"

type statusFlags struct {
	DefaultStatusOnCreation bool `json:"defaultStatusOnCreation"`
	FillEmptyStartDate      bool `json:"fillEmptyStartDate"`
	SetStartDate            bool `json:"setStartDate"`
	FillEmptyFinishDate     bool `json:"fillEmptyFinishDate"`
	SetFinishDate           bool `json:"setFinishDate"`
}

type addRequest struct {
	ID       string      `json:"id"`
	Name     string      `json:"name"`
	HexColor string      `json:"hexColor"`
	Index    uint        `json:"index"`
	Flags    statusFlags `json:"flags"`
}

type updateRequest struct {
	ID       string      `json:"id"`
	Name     string      `json:"name"`
	HexColor string      `json:"hexColor"`
	Index    uint        `json:"index"`
	Flags    statusFlags `json:"flags"`
}

type filterRequest struct {
	Name *string `json:"name"`
}

type searchRequest struct {
	Pager  handlers.PagerRequest `json:"pager"`
	Order  handlers.OrderRequest `json:"order"`
	Filter *filterRequest        `json:"filter"`
}

type TaskStatusResponse struct {
	ID       string      `json:"id"`
	Name     string      `json:"name"`
	HexColor string      `json:"hexColor"`
	Index    uint        `json:"index"`
	Flags    statusFlags `json:"flags"`
}

type searchResponse struct {
	TaskStatuses []TaskStatusResponse   `json:"taskStatuses"`
	Pager        handlers.PagerResponse `json:"pager"`
}
